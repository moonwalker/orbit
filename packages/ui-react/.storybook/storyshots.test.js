import initStoryshots, { Stories2SnapsConverter } from '@storybook/addon-storyshots';
import { render } from '@testing-library/react';

initStoryshots({
  asyncJest: true,
  test: ({ story, context, done }) => {
    const converter = new Stories2SnapsConverter();
    const snapshotFilename = converter.getSnapshotFileName(context);
    const storyElement = story.render();

    const { container } = render(storyElement);

    setTimeout(() => {
      if (snapshotFilename) {
        expect(container.firstChild).toMatchSpecificSnapshot(snapshotFilename);
      }

      done();
    }, 1);
  }
});
