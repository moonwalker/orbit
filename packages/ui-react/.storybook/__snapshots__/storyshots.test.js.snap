// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Atoms/Button default 1`] = `
<button
  className="ui-button"
  onClick={[Function]}
>
  Call to action
</button>
`;

exports[`Storyshots Components/Atoms/Button with clear 1`] = `
<button
  className="ui-button ui-button--clear"
  onClick={[Function]}
>
  Call to action
</button>
`;

exports[`Storyshots Components/Atoms/Button with custom content 1`] = `
<button
  className="ui-button"
  onClick={[Function]}
>
  <em
    className="custom ui-button__content"
    size={null}
  >
    Call
  </em>
  <span
    className="ui-button__content"
  >
    to action
  </span>
</button>
`;

exports[`Storyshots Components/Atoms/Button with different icon size 1`] = `
<button
  className="ui-button"
  onClick={[Function]}
>
  <i
    className="ui-icon ui-button__content ui-icon--large"
  >
    backup
  </i>
  <span
    className="ui-button__content"
  >
    Call to action
  </span>
</button>
`;

exports[`Storyshots Components/Atoms/Button with kind 1`] = `
<button
  className="ui-button ui-button--danger"
  onClick={[Function]}
>
  Call to action
</button>
`;

exports[`Storyshots Components/Atoms/Button with left icon 1`] = `
<button
  className="ui-button"
  onClick={[Function]}
>
  <i
    className="ui-icon ui-button__content"
  >
    backup
  </i>
  <span
    className="ui-button__content"
  >
    Call to action
  </span>
</button>
`;

exports[`Storyshots Components/Atoms/Icon default 1`] = `
<i
  className="ui-icon"
>
  face
</i>
`;

exports[`Storyshots Components/Atoms/Input default 1`] = `
<input
  className="ui-input ui-input--medium"
  onChange={[Function]}
  placeholder="Enter text here"
/>
`;

exports[`Storyshots Components/Atoms/Input invalid 1`] = `
<input
  className="ui-input ui-input--medium ui-input--invalid"
  onChange={[Function]}
  placeholder="Enter text here"
/>
`;

exports[`Storyshots Components/Atoms/Input valid 1`] = `
<input
  className="ui-input ui-input--medium ui-input--valid"
  onChange={[Function]}
  placeholder="Enter text here"
/>
`;

exports[`Storyshots Components/Atoms/Label default 1`] = `
<label
  className="ui-label"
>
  Username
</label>
`;

exports[`Storyshots Components/Atoms/Label with size variation 1`] = `
<label
  className="ui-label ui-label--large"
>
  Username
</label>
`;

exports[`Storyshots Components/Atoms/Select default 1`] = `
<select
  className="ui-select ui-select--medium"
  onChange={[Function]}
>
  <option>
    Option A
  </option>
  <option>
    Option B
  </option>
  <option>
    Option C
  </option>
</select>
`;

exports[`Storyshots Components/Atoms/Select invalid 1`] = `
<select
  className="ui-select ui-select--medium ui-select--invalid"
  onChange={[Function]}
>
  <option>
    Option A
  </option>
  <option>
    Option B
  </option>
  <option>
    Option C
  </option>
</select>
`;

exports[`Storyshots Components/Atoms/Select valid 1`] = `
<select
  className="ui-select ui-select--medium ui-select--valid"
  onChange={[Function]}
>
  <option>
    Option A
  </option>
  <option>
    Option B
  </option>
  <option>
    Option C
  </option>
</select>
`;

exports[`Storyshots Components/Atoms/Select with options attributes 1`] = `
<select
  className="ui-select ui-select--medium"
  onChange={[Function]}
>
  <option
    value={0}
  >
    Option A
  </option>
  <option
    selected={true}
    value={1}
  >
    Option B
  </option>
  <option
    value={2}
  >
    Option C
  </option>
</select>
`;

exports[`Storyshots Components/Atoms/Select with size variation 1`] = `
<select
  className="ui-select ui-select--large"
  onChange={[Function]}
>
  <option>
    Option A
  </option>
  <option>
    Option B
  </option>
  <option>
    Option C
  </option>
</select>
`;

exports[`Storyshots Components/Atoms/Spinner default 1`] = `
<span
  className="ui-spinner"
/>
`;

exports[`Storyshots Components/Atoms/Spinner with custom size 1`] = `
<span
  className="ui-spinner ui-spinner--large"
/>
`;

exports[`Storyshots Components/Atoms/Textarea default 1`] = `
<textarea
  className="ui-textarea ui-textarea--medium"
  onChange={[Function]}
  placeholder="Enter text here"
/>
`;

exports[`Storyshots Components/Atoms/Textarea invalid 1`] = `
<textarea
  className="ui-textarea ui-textarea--medium ui-textarea--invalid"
  onChange={[Function]}
  placeholder="Enter text here"
/>
`;

exports[`Storyshots Components/Atoms/Textarea valid 1`] = `
<textarea
  className="ui-textarea ui-textarea--medium ui-textarea--valid"
  onChange={[Function]}
  placeholder="Enter text here"
/>
`;

exports[`Storyshots Examples/Contact Form default 1`] = `
<div
  style={
    Object {
      "margin": "0 auto",
      "maxWidth": "380px",
      "padding": "1rem",
    }
  }
>
  <form
    className="contact-form"
    onSubmit={[Function]}
  >
    <div
      className="contact-form__field"
    >
      <label
        className="ui-label contact-form__field__label"
      >
        First name
      </label>
      <input
        className="ui-input"
        name="firstname"
        placeholder="First name"
      />
    </div>
    <div
      className="contact-form__field"
    >
      <label
        className="ui-label contact-form__field__label"
      >
        Last name
      </label>
      <input
        className="ui-input"
        name="lastname"
        placeholder="Last name"
      />
    </div>
    <div
      className="contact-form__field"
    >
      <label
        className="ui-label contact-form__field__label"
      >
        Country
      </label>
      <select
        className="ui-select"
        name="country"
      >
        <option>
          Afghanistan
        </option>
        <option>
          Albania
        </option>
        <option>
          Algeria
        </option>
        <option>
          Andorra
        </option>
        <option>
          Angola
        </option>
        <option>
          Antigua and Barbuda
        </option>
        <option>
          Argentina
        </option>
        <option>
          Armenia
        </option>
        <option>
          Aruba
        </option>
        <option>
          Australia
        </option>
        <option>
          Austria
        </option>
        <option>
          Azerbaijan
        </option>
      </select>
    </div>
    <div
      className="contact-form__field"
    >
      <label
        className="ui-label contact-form__field__label"
      >
        Message
      </label>
      <textarea
        className="ui-textarea"
        name="message"
        placeholder="Enter your message"
        rows="5"
      />
    </div>
    <div
      className="contact-form__actions"
    >
      <button
        className="ui-button"
      >
        Send
      </button>
    </div>
  </form>
</div>
`;

exports[`Storyshots Examples/SearchField default 1`] = `
<div
  style={
    Object {
      "margin": "0 auto",
      "maxWidth": "380px",
      "padding": "1rem",
    }
  }
>
  <div
    className="search-field"
  >
    <input
      aria-activedescendant={null}
      aria-autocomplete="list"
      aria-expanded={false}
      autoComplete="off"
      className="ui-input search-field__input ui-input--large"
      id="downshift-0-input"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Search for something"
      role="combobox"
      value=""
    />
    <i
      className="ui-icon search-field__icon ui-icon--large"
    >
      search
    </i>
  </div>
</div>
`;

exports[`Storyshots Examples/SearchField with predefined query 1`] = `
<div
  style={
    Object {
      "margin": "0 auto",
      "maxWidth": "380px",
      "padding": "1rem",
    }
  }
>
  <div
    className="search-field"
  >
    <input
      aria-activedescendant={null}
      aria-autocomplete="list"
      aria-expanded={true}
      autoComplete="off"
      className="ui-input search-field__input ui-input--large"
      id="downshift-1-input"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Search for something"
      role="combobox"
      value="Lo"
    />
    <i
      className="ui-icon search-field__icon ui-icon--large"
    >
      search
    </i>
    <div
      className="search-field__results"
    >
      <div
        className="search-field__item"
        id="downshift-1-item-0"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseMove={[Function]}
      >
        Lottie Ramirez
      </div>
      <div
        className="search-field__item"
        id="downshift-1-item-1"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseMove={[Function]}
      >
        Kirkland Cook
      </div>
      <div
        className="search-field__item"
        id="downshift-1-item-2"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseMove={[Function]}
      >
        Marcie Adkins
      </div>
    </div>
  </div>
</div>
`;
